pkg/common/cns-lib/volume/manager.go

```diff
@@ -56,7 +56,7 @@ type Manager interface {
 	// CreateVolume creates a new volume given its spec.
 	CreateVolume(ctx context.Context, spec *cnstypes.CnsVolumeCreateSpec) (*CnsVolumeInfo, error)
 	// AttachVolume attaches a volume to a virtual machine given the spec.
-	AttachVolume(ctx context.Context, vm *cnsvsphere.VirtualMachine, volumeID string) (string, error)
+	AttachVolume(ctx context.Context, vm *cnsvsphere.VirtualMachine, volumeID string, checkNVMeController bool) (string, error)
```

Comment:
Introducing this bool to make sure NVMe controller check is only for supervisor cluster

Reply:
How about adding this as a parameter to `defaultManager`? You'd just need to pass this parameter once during init based on cluster flavor.

Reply:
I will follow-up with this suggestion in the next iteration. Created an issue to track the same: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/913